src/walk.ts

```diff
@@ -25,18 +25,34 @@ export type Options = {
  * @returns An array of absolute paths of all parent directories.
  */
 export function up(base: string, options?: Options): string[] {
-	let { stop, cwd } = options || {};
+	if (typeof options?.stop === 'undefined') {
```

Comment:
The rest of this file is simply this:

```js
// pseudo code, to be clear
if (stop !== undefined) {
  while (tmp !== stop) {
    // push
  }
} else {
  while (tmp !== prev) {
   // push
  }
}
```

Again, I had this. There was performance loss when moving from `while (true)` to adding a condition (because obviously) but it was predominantly regained by consolidating while loops & making it `while (root || ...)` where `root = true` for no options.stop. Effectively `while (true || ...)` is the same as `while (true)` because it never has to evaluate the other side of the pipes... and in worst case (having a stop condition), it has to pass thru a `while (false || <eval>)` which is insignificant cost.